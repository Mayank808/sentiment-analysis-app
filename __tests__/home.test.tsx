import { render, screen } from "@testing-library/react";
import { vi } from "vitest";
import Home from "../app/page";

vi.mock("@clerk/nextjs", () => {
  // Create an mockedFunctions object to match the functions we are importing from the @nextjs/clerk package in the ClerkComponent component.
  const mockedFunctions = {
    auth: () =>
      new Promise((resolve) => resolve({ userId: "test_user_id_vite" })),
    ClerkProvider: ({ children }) => <div>{children}</div>,
    useUser: () => ({
      isSignedIn: true,
      user: {
        id: "test_user_id_vite",
        fullName: "test_user vitest",
      },
    }),
  };

  return mockedFunctions;
});

// vi.mock("@clerk/nextjs", () => {
//   // Create an mockedFunctions object to match the functions we are importing from the @nextjs/clerk package in the ClerkComponent component.
//   const mockedFunctions = {
//     auth: () => new Promise((resolve) => resolve({ userId: null })),
//     ClerkProvider: ({ children }) => <div>{children}</div>,
//     useUser: () => ({
//       isSignedIn: false,
//       user: null,
//     }),
//   };

//   return mockedFunctions;
// });

describe("Home UI tests", () => {
  vi.mock("next/font/google", () => {
    return {
      Inter: () => ({ className: "inter" }),
    };
  });

  test(`Home Snapshot test`, async () => {
    const home = render(await Home());
    expect(home).toMatchSnapshot();
  });

  test(`Testing Home page render basic text`, async () => {
    const screen = render(await Home());

    expect(screen.getByText("ScribeMe")).toBeTruthy();
    expect(screen.getByText("Get started")).toBeTruthy();

    const childComponent = screen.getByTestId("typewriter-wrapper");
    expect(childComponent).toBeInTheDocument();
  });

  test(`Testing dynamic routing when user is logged in`, async () => {
    const url = "/scribe";
    const screen = render(await Home());
    const dynamicLink = screen.getByTestId("routing");
    expect(dynamicLink).toHaveAttribute("href", url);
  });

  // test(`Testing dynamic routing when user is not signed in`, async () => {
  //   const url = "/new-user";
  //   const screen = render(await Home());
  //   const dynamicLink = screen.getByTestId("routing");
  //   expect(dynamicLink).toHaveAttribute("href", url);
  // });
});
