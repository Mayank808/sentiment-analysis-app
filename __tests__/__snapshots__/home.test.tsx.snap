// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dashboard UI tests > Home Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-screen h-screen bg-background flex justify-center items-center text-white"
      >
        <div
          class="w-full max-w-[600px] mx-auto text-center sm:text-left"
        >
          <h1
            class="text-6xl"
          >
            ScribeMe
          </h1>
          <div
            class="pt-3 text-2xl text-white/60"
          >
            <div
              class="Typewriter"
              data-testid="typewriter-wrapper"
            >
              <span
                class="Typewriter__wrapper"
              />
              <span
                class="Typewriter__cursor"
              >
                |
              </span>
            </div>
          </div>
          <div
            class="my-4"
          >
            <a
              data-testid="routing"
              href="/scribe"
            >
              <button
                class="bg-blue px-4 py-2 rounded-lg text-xl"
              >
                Get started
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-screen h-screen bg-background flex justify-center items-center text-white"
    >
      <div
        class="w-full max-w-[600px] mx-auto text-center sm:text-left"
      >
        <h1
          class="text-6xl"
        >
          ScribeMe
        </h1>
        <div
          class="pt-3 text-2xl text-white/60"
        >
          <div
            class="Typewriter"
            data-testid="typewriter-wrapper"
          >
            <span
              class="Typewriter__wrapper"
            />
            <span
              class="Typewriter__cursor"
            >
              |
            </span>
          </div>
        </div>
        <div
          class="my-4"
        >
          <a
            data-testid="routing"
            href="/scribe"
          >
            <button
              class="bg-blue px-4 py-2 rounded-lg text-xl"
            >
              Get started
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-screen h-screen bg-background flex justify-center items-center text-white"
      >
        <div
          class="w-full max-w-[600px] mx-auto text-center sm:text-left"
        >
          <h1
            class="text-6xl"
          >
            ScribeMe
          </h1>
          <div
            class="pt-3 text-2xl text-white/60"
          >
            <div
              class="Typewriter"
              data-testid="typewriter-wrapper"
            >
              <span
                class="Typewriter__wrapper"
              />
              <span
                class="Typewriter__cursor"
              >
                |
              </span>
            </div>
          </div>
          <div
            class="my-4"
          >
            <a
              data-testid="routing"
              href="/scribe"
            >
              <button
                class="bg-blue px-4 py-2 rounded-lg text-xl"
              >
                Get started
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-screen h-screen bg-background flex justify-center items-center text-white"
    >
      <div
        class="w-full max-w-[600px] mx-auto text-center sm:text-left"
      >
        <h1
          class="text-6xl"
        >
          ScribeMe
        </h1>
        <div
          class="pt-3 text-2xl text-white/60"
        >
          <div
            class="Typewriter"
            data-testid="typewriter-wrapper"
          >
            <span
              class="Typewriter__wrapper"
            />
            <span
              class="Typewriter__cursor"
            >
              |
            </span>
          </div>
        </div>
        <div
          class="my-4"
        >
          <a
            data-testid="routing"
            href="/scribe"
          >
            <button
              class="bg-blue px-4 py-2 rounded-lg text-xl"
            >
              Get started
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home UI tests > Home Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-screen h-screen bg-background flex justify-center items-center text-white"
      >
        <div
          class="w-full max-w-[600px] mx-auto text-center sm:text-left"
        >
          <h1
            class="text-6xl"
          >
            ScribeMe
          </h1>
          <div
            class="pt-3 text-2xl text-white/60"
          >
            <div
              class="Typewriter"
              data-testid="typewriter-wrapper"
            >
              <span
                class="Typewriter__wrapper"
              />
              <span
                class="Typewriter__cursor"
              >
                |
              </span>
            </div>
          </div>
          <div
            class="my-4"
          >
            <a
              data-testid="routing"
              href="/scribe"
            >
              <button
                class="bg-blue px-4 py-2 rounded-lg text-xl"
              >
                Get started
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-screen h-screen bg-background flex justify-center items-center text-white"
    >
      <div
        class="w-full max-w-[600px] mx-auto text-center sm:text-left"
      >
        <h1
          class="text-6xl"
        >
          ScribeMe
        </h1>
        <div
          class="pt-3 text-2xl text-white/60"
        >
          <div
            class="Typewriter"
            data-testid="typewriter-wrapper"
          >
            <span
              class="Typewriter__wrapper"
            />
            <span
              class="Typewriter__cursor"
            >
              |
            </span>
          </div>
        </div>
        <div
          class="my-4"
        >
          <a
            data-testid="routing"
            href="/scribe"
          >
            <button
              class="bg-blue px-4 py-2 rounded-lg text-xl"
            >
              Get started
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
